sdk/core/src/services/purl2cpe/service.rs

```diff
@@ -0,0 +1,486 @@
+    pub(crate) store: Arc<Store>,
+}
+
+#[async_trait]
+impl MongoService<Purl2Cpes> for Purl2CpeService {
+    /// Getter for the store
+    fn store(&self) -> Arc<Store> {
+        self.store.clone()
+    }
+
+    /// Insert a document into a [Collection].
+    async fn insert<'a>(&self, doc: &mut Purl2Cpes) -> Result<(), PlatformError> {
+        if doc.id.is_empty() {
+            return Err(PlatformError::Mongo(String::from(
+                "==> purl2cpe::id(purl)::empty",
+            )));
+        }
+
+        if doc.cpes.is_empty() {
+            return Err(PlatformError::Mongo(String::from(
+                "==> purl2cpe::cpes::empty",
+            )));
+        }
+
+        self.store.insert(doc).await?;
+        Ok(())
+    }
+}
+
+fn find_cpe(version: String, cpes: Vec<String>) -> Option<String> {
+    let mut final_cpe = None::<String>;
+    let all_versions = String::from("*");
+    let no_version = String::from("-");
+    for cpe_str in cpes {
+        let cpe = Cpe::new(cpe_str);
+
+        // If this conditional evaluates to true, we have exactly
+        // what we want and we can short circuit the loop
+        if cpe.version.eq(&version) {
+            final_cpe = Some(cpe.get_string());
+            break;
+        }
+
+        // If the default is found, we load that up but keep looping
+        if cpe.version.eq(&all_versions) || cpe.version.eq(&no_version) {
+            final_cpe = Some(cpe.get_string());
+        }
+    }
+
+    final_cpe
+}
+
+impl Purl2CpeService {
+    /// Creates a new Purl2CpeService
+    pub fn new(store: Arc<Store>) -> Self {
+        Purl2CpeService { store }
+    }
+
+    /// Method to update a Package document in the store with a cpe
+    pub async fn update_package_with_cpe(&self, id: String, cpe: String) -> Result<(), Error> {
+        let mut package_doc = match self
+            .store
+            .find::<Package>(id.as_str())
+            .await
+            .map_err(Error::Mongo)?
+        {
+            None => {
+                return Err(Error::Purl2Cpe(format!(
+                    "Unable to find Package document in the data store with id: {}",
+                    id
+                )))
+            }
+            Some(package_doc) => package_doc,
+        };
+
+        package_doc.cpe = Some(cpe);
+        self.store
+            .update::<Package>(&package_doc)
+            .await
+            .map_err(Error::Mongo)?;
+
+        Ok(())
+    }
+
+    /// Method to get a cpe for a purl
+    pub async fn get_cpe(&self, full_purl: String) -> Result<Option<String>, Error> {
+        fn error_result(msg: &str) -> Result<Option<String>, Error> {
+            Err(Error::Purl2Cpe(String::from(msg)))
+        }
+
+        let at_sign = '@';
+
+        if full_purl.is_empty() {
+            return error_result("==> unable to process, purl is empty");
+        }
+
+        return if full_purl.contains(at_sign) {
+            let full_purl: Vec<&str> = full_purl.split(at_sign).collect();
```

Comment:
Not a blocker, but these two if..else blocks might be easier to read by future us if they were moved into functions with intention revealing names.

Reply:
Good idea, I'll do that